### PR TITLE
Default maxRequests to Integer.MAX_VALUE rather than 1000

### PR DESCRIPTION
**Pull request by mzalimeni**
Default maxRequests to Integer.MAX_VALUE rather than 1000 to allow it to be used in combination with the duration setting (currently, duration causes an override of any maxRequests value to Integer.MAX_VALUE)

### CODE REVIEW

**Comment by WamBamBoozle** [comment]
We need to set the default in src/main/scala/com/twitter/parrot/config/ParrotFeederConfig.scala as well.

**Comment by mzalimeni** [comment]
Oops - fixing failing test.

**Comment by mzalimeni** [comment]
Let me know your thoughts on this one. I believe things are good to go.

File: src/main/scala/com/twitter/parrot/feeder/ParrotFeeder.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ class ParrotFeeder(config: ParrotFeederConfig) extends Service {
   private[this] val log = Logger.get(getClass.getName)
   val requestsRead = new AtomicLong(0)
   @volatile
-  private[this] var state = FeederState.RUNNING
+  protected[this] var state = FeederState.RUNNING
```
Comment (WamBamBoozle) [inline src/main/scala/com/twitter/parrot/feeder/ParrotFeeder.scala]: Is the expression "protected[this]" meaningful?

File: src/main/scala/com/twitter/parrot/util/RemoteParrot.scala [hunk 2]
```diff
@@ -117,7 +117,7 @@ class RemoteParrot(val name: String,
 
   def shutdown() {
     consumer.shutdown
-    waitFor(client.shutdown())
+    waitFor(client.shutdown(), Duration.Top)
```
Comment (WamBamBoozle) [inline src/main/scala/com/twitter/parrot/util/RemoteParrot.scala]: Adding Duration.Top does not change anything.

File: src/main/scala/com/twitter/parrot/util/RemoteParrot.scala [hunk 3]
```diff
@@ -156,8 +156,8 @@ class RemoteParrot(val name: String,
     (service, client)
   }
 
-  private[this] def waitFor[A](future: Future[A]): A = {
-    future.get(finagleTimeout) match {
+  private[this] def waitFor[A](future: Future[A], timeout: Duration = finagleTimeout): A = {
```
Comment (WamBamBoozle) [inline src/main/scala/com/twitter/parrot/util/RemoteParrot.scala]: Just don't use a timeout. The default behavior of future.get is to wait forever.

**Comment by WamBamBoozle** [comment]
Looks good to me. Why'd you close it?

On Tue, Jan 13, 2015 at 11:18 AM, Michael Zalimeni <notifications@github.com

File: src/main/scala/com/twitter/parrot/feeder/ParrotFeeder.scala (hunk 1 above)
Comment (mzalimeni) [inline src/main/scala/com/twitter/parrot/feeder/ParrotFeeder.scala]: I ended up using an explicit state change check in EndToEndSpec, so I needed to bump up to protected access for that property in an anonymous extension of ParrotFeeder

**Comment by mzalimeni** [comment]
Thanks! I think GitHub actually posted my last comment to you (unintentional) when I closed it. I'd intended to pull out one of my commits - switching poller and cluster order, which I noticed yesterday was taking away helpful feedback for a trivial log-quieting benefit - then rebase and make a new request to keep your logs cleaner.

Mind if I re-submit? Sorry for the confusion.

**Comment by WamBamBoozle** [comment]
no worries

Please re-submit.

On Tue, Jan 13, 2015 at 11:44 AM, Michael Zalimeni <notifications@github.com

**Pull request closed without merging**